validation/poststart_fail.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
```

Comment:
No need to set `util.LifecycleActionDelete` or the `PreDelete` code below, because [the `start` call should exit nonzero][1] after the hook failure.  That means `RuntimeLifecycleValidate` will exit [here][2], before any of its delete-related code fires.

[1]: https://github.com/opencontainers/runtime-tools/blob/3eba113f4e36cca8795c9835a437a0f12b8fdb89/docs/command-line-interface.md#start
[2]: https://github.com/opencontainers/runtime-tools/blob/3eba113f4e36cca8795c9835a437a0f12b8fdb89/validation/util/test.go#L319

Reply:
https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#lifecycle 
In poststart fail test, it will continue to work. 'start' should not exit.

Also with 'ActionDelete' set, even there is no chance to get 'r.Delete', the container will be deleted in the 'defer' function.
https://github.com/opencontainers/runtime-tools/blob/3eba113f4e36cca8795c9835a437a0f12b8fdb89/validation/util/test.go#L294